models/intermediate/int_amazon_selling_partner__date_spine.sql

```diff
@@ -0,0 +1,36 @@
+with spine as (
+
+    {% if execute and flags.WHICH in ('run', 'build') %}
+
+    {%- set first_date_query %}
+    select 
+        coalesce(
+            min(cast(purchase_date as date)), 
+            cast({{ dbt.dateadd("month", -1, "current_date") }} as date)
+            ) as min_date
+    from {{ ref('stg_amazon_selling_partner__orders') }} -- Open Q: should we use items instead, since a shop will likely release items prior to having sales
```

Comment:
That's a good point, you can't have an order unless you have an item to sell.

Reply:
As a result of this, wouldn't this model need to be associated with the corresponding variable to turn on/off?

Reply:
removed this model